### PR TITLE
Add the "Deleted" user

### PR DESCRIPTION
**Pull request by allenwq**
Also renames `#system?` to `#default?`, this method now indicates whether the account is a builtin default account.

### CODE REVIEW

**Comment by lowjoel** [comment]
Call it `built_in?`

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -21,17 +21,26 @@
...
+        system_user = User.system
+        expect(system_user.password).to be_nil
+        expect(system_user.email).to be_nil
+        expect(system_user.default?).to be_truthy
       end
     end
 
-    describe '#system?' do
+    describe '.deleted' do
+      it 'returns the system user' do
```
Comment (lowjoel) [inline spec/models/user_spec.rb]: deleted user

File: db/seeds.rb [hunk 2]
```diff
@@ -13,13 +13,17 @@
...
-  unless user
+  # Create the Coursemology default user accounts.
+  unless User.exists?(User::SYSTEM_USER_ID)
     User.create!(id: User::SYSTEM_USER_ID, name: 'System')
   end
+  unless User.exists?(User::DELETED_USER_ID)
+    User.new(id: User::DELETED_USER_ID, name: 'Deleted').save!
+  end
+  # Sequence were outdated when we specify the id.
+  ActiveRecord::Base.connection.reset_pk_sequence!(User.table_name)
```
Comment (allenwq) [inline db/seeds.rb]: @lowjoel I did't find other ways of making manually assign id work, looks not pretty though.

File: spec/models/user_spec.rb (hunk 1 above)
Comment (allenwq) [inline spec/models/user_spec.rb]: Oops, thanks !

**Comment by allenwq** [comment]
@lowjoel Renamed to built_in

**Comment by allenwq** [comment]
@lowjoel I didn't notice that the test failed, any comments now ?

File: db/seeds.rb (hunk 2 above)
Comment (lowjoel) [inline db/seeds.rb]: why not make the id -1 then?

Comment (allenwq) [inline db/seeds.rb]: meaning ?  id 0 is used by system user already

Comment (lowjoel) [inline db/seeds.rb]: yeah, so 0 is system, -1 is deleted. negative numbers should be evidently clear it's... not normal.

Comment (allenwq) [inline db/seeds.rb]: Okay, using negative PK now. Btw, is it a convention to use negative id for preserved records ?

**Pull request merged**